js/screens/Competition/CompetitionContainer.js

```diff
@@ -10,13 +20,87 @@ class CompetitionContainer extends Component {
     ...headerBarStyle(navigation)
   });
 
+  constructor() {
+    super();
+    this.findBestMatch = this.findBestMatch.bind(this);
+    this.findOtherMatches = this.findOtherMatches.bind(this);
+    this.addUser = this.addUser.bind(this);
+  }
+
+  async findBestMatch(teamSize) {
+    await this.props.dispatch(fetchUserWorkstyle());
+    //TODO: change the user score to dynamic values
+    await this.props.dispatch(getUserScore([4, 1, 5]));
+
+    await this.props.dispatch(
+      fetchBestMatch(
+        this.props.userWorkstyle,
+        this.props.userScore,
+        this.props.navigation.state.params.data.id,
+        teamSize
+      )
+    );
+  }
+
+  async findOtherMatches() {
+    if (
+      this.props.userWorkstyle === undefined ||
+      this.props.userScore === undefined
+    ) {
+      await this.props.dispatch(fetchUserWorkstyle());
+      //TODO: change the user score to dynamic values
+      await this.props.dispatch(getUserScore([1, 4, 5]));
+    }
+
+    await this.props.dispatch(
+      fetchOtherMatches(
+        this.props.userWorkstyle,
+        this.props.userScore,
+        this.props.navigation.state.params.data.id,
+        false
+      )
+    );
+  }
+
+  async addUser() {
+    //TODO: change the user score to dynamic values
+    await this.props.dispatch(getUserScore([1, 4, 5]));
+
+    const uid = await AsyncStorage.getItem("user");
+    this.props.dispatch(
+      addUserToTeam(this.props.userScore, this.props.teamId, uid)
+    );
+  }
+
   render() {
-    return <Competition data={this.props.navigation.state.params.data} />;
+    return (
+      <Competition
+        data={this.props.navigation.state.params.data}
+        findBestMatch={this.findBestMatch}
+        findOtherMatches={this.findOtherMatches}
+        addUser={this.addUser}
+      />
+    );
   }
 }
 
-export default CompetitionContainer;
+const mapStateToProps = state => ({
+  teams: state.teams.otherMatches,
+  userScore: state.user.score,
+  userWorkstyle: state.user.workstyle,
+  teamId: state.teams.bestMatch.id
+});
+
+export default connect(mapStateToProps)(CompetitionContainer);
 
 CompetitionContainer.propTypes = {
-  navigation: PropTypes.object.isRequired
+  navigation: PropTypes.object.isRequired,
+  dispatch: PropTypes.func.isRequired,
+  userScore: PropTypes.array.isRequired,
+  userWorkstyle: PropTypes.string.isRequired,
+  teamId: PropTypes.string.isRequired
+};
+
+CompetitionContainer.defaultProps = {
+  teamId: ""
```

Comment:
the teamId is only passed in when the user decides to join the team, hence there will be no teamId when the app loads. The default prop is to prevent error.

Reply:
in that case, you do not have to say .isRequired, passing default prop is enough.